### PR TITLE
migrate updater to use jinja templating

### PR DESCRIPTION
**Pull request by gdams**

### CODE REVIEW

**Review by karianna — COMMENTED** [review]
Seems to be additions like:

amd64|i386:x86-64) \
         ESUM='7b175dbe0d6e3c9c23b6ed96449b018308d8fc94a5ecd9c0df8b8bc376c3c18a'; \
         BINARY_URL='https://github.com/adoptium/temurin17-binaries/releases/download/jdk-17.0.9%2B9/OpenJDK17U-jdk_x64_linux_hotspot_17.0.9_9.tar.gz'; \
         ;; \
         
         I suspect that should be split out?

**Comment by gdams** [comment]
> Seems to be additions like:
> 
> 
> 
> amd64|i386:x86-64) \
> 
>          ESUM='7b175dbe0d6e3c9c23b6ed96449b018308d8fc94a5ecd9c0df8b8bc376c3c18a'; \
> 
>          BINARY_URL='https://github.com/adoptium/temurin17-binaries/releases/download/jdk-17.0.9%2B9/OpenJDK17U-jdk_x64_linux_hotspot_17.0.9_9.tar.gz'; \
> 
>          ;; \
> 
>          
> 
>          I suspect that should be split out?

I felt unsure what the best path forward was here. I think it makes sense to remove the dockerfile updates completely to make the PR easier to review

File: docker_templates/partials/header.j2 [hunk 1]
```diff
@@ -0,0 +1,5 @@
+ENV JAVA_HOME /opt/java/openjdk
+ENV PATH $JAVA_HOME/bin:$PATH
+
+# Default to UTF-8 file.encoding
+ENV LANG='en_US.UTF-8' LANGUAGE='en_US:en' LC_ALL='en_US.UTF-8'
```
Comment (tellison) [inline docker_templates/partials/header.j2:5]: This partials file might better be called something like `nix-env.j2`?

File: generate_dockerfiles.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
+import os
```
Comment (tellison) [inline generate_dockerfiles.py:1]: 1. Files of this length/complexity should have a source file header
2. Would be good to have some diagnostic logging in this file to help track down any misbehavior in action

**Review by tellison — APPROVED** [review]
Some suggestions.

File: docker_templates/partials/header.j2 (hunk 1 above)
Comment (gdams) [reply docker_templates/partials/header.j2]: done

File: generate_dockerfiles.py (hunk 2 above)
Comment (gdams) [reply generate_dockerfiles.py:22]: Added the source file header

**Review by jerboaa — APPROVED** [review]
Seems OK, though fairly complex.

**Pull request merged**